cfme/common/vm.py

```diff
@@ -63,6 +63,72 @@ def all_types(template=False):
     return all_types
 
 
+def do_set_retirement_date(view, when=None, offset=None, warn=None, multiple_vms=False):
```

Comment:
Rather than taking the view, I think this should take the object and do the navigation within the method.

Reply:
I think I understand why you've moved this into a completely separate function -- so that you can set the retirement date of multiple entities, correct?

In my view, in order for this to mesh with our framework -- this should be a method of vm/instance and also a (filtered) collection of those entities. You could even make it a function in `cfme/utils` and then call it from the method of the instance/vm/collection. This will avoid having to redefine the calling sequence each time like you do in `set_retirement_date_multiple` in `test_retirement.py`

Reply:
I've rewritten this module-level method (now called set_retirement_date) so that it accepts either a VM/Instance object or a list of them. If  a list, then the method sets the retirement date from the All VMs/Instances page. Otherwise it navigates to the VM's/Instance's details page and sets it there.